src/shared/compositeKeyMap.ts

```diff
@@ -0,0 +1,38 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Lewis Daly <lewisd@crosslaketech.com>
+ --------------
+ ******/
+
+/**
+ * @class CompositeKeyMap
+ * @description CompositeKeyMap is a special extension of Map, which will stringify the `key` value passed in
+ *   This allows us to nicely and easily set composite keys
+ */
+export default class CompositeKeyMap<T, U> extends Map {
+  public get (key: T): U {
+    return super.get(JSON.stringify(key))
```

Comment:
JSON.stringify isn't returning a canonical string. 
try this semantically identical keys:
```JSON.stringify({a:1,b:2}) !== JSON.stringify({b:2, a:1})```
maybe it is better to use https://github.com/mirkokiefer/canonical-json

Reply:
Ah good point.